receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo.go

```diff
@@ -90,7 +94,7 @@ func (e *EcsInfo) GetClusterName() string {
 type ecsInfoOption func(*EcsInfo)
 
 // New creates a k8sApiServer which can generate cluster-level metrics
-func NewECSInfo(refreshInterval time.Duration, hostIPProvider hostIPProvider, host component.Host, settings component.TelemetrySettings, options ...ecsInfoOption) (*EcsInfo, error) {
+func NewECSInfo(refreshInterval time.Duration, clusterName string, hostIPProvider hostIPProvider, host component.Host, settings component.TelemetrySettings, options ...ecsInfoOption) (*EcsInfo, error) {
```

Comment:
nit: It looks like this constructor uses the options pattern. Seeing as there are so many fields and `clusterName` can be optional maybe we could make an `ecsInfoOption` for it. Not really sure how or if the options are being used since `ecsInfoOption` is private.

```go
func WithClusterName(name string) ecsInfoOption {
  return func(info *EcsInfo) {
      info.clusterName = name
  }
}
```

Reply:
Yeah.. i might have to expose `ecsInfoOption` although I dont see an issue with that. Ill give it a go.